### PR TITLE
DB records abstraction

### PR DESCRIPTION
**Pull request by eddowh**
Covers #43

### CODE REVIEW

File: openkongqi/records/base.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    def create_dsn(self, settings):
+        """Create a data source name (DNS) given a settings dict.
+
+        .. warning:: This method has to be overwritten
+        """
+        raise NotImplementedError
+
+    def create_db_session(self, settings):
+        """Create a SQLAlchemy session with engine binded."""
```
Comment (hrbonz) [inline openkongqi/records/base.py]: make it generic, it's the base class

File: openkongqi/records/base.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    as parent of any database records wrapper class.
+
+    The children wrapper class will have to define custom methods,
+    for creating the database connection.
+    """
+
+    def __init__(self, settings, *args, **kwargs):
+        self._db = self.create_db_session(settings)
+
+    def create_dsn(self, settings):
```
Comment (hrbonz) [inline openkongqi/records/base.py]: move this method to only sqlalchemy and get it to return a URL instance.

**Comment by eddowh** [comment]
Tried it out, works on `sqlite3`. Still need to test other engines.

File: openkongqi/db.py [hunk 3]
```diff
@@ -1,21 +1,11 @@
...
+from .records.sqlalchemy_wrapper import Base
 
 
 def db_init():
-    """Create the database"""
-    import openkongqi.models
-    Base.metadata.create_all(bind=engine)
+    """Create the database."""
+    # FIXME - temporary fix specific to SQLAlchemy
+    Base.metadata.create_all(bind=recsdb.get_engine())
```
Comment (hrbonz) [inline openkongqi/db.py]: Create a `db_init` method on `BaseRecordsWrapper` that should take over this, `.db` can now be simply removed.
Create the same `db_init` method on `BaseStatusWrapper` which for redis would do nothing.

**Comment by hrbonz** [comment]
Rename `.records.sqlalchmey_wrapper` to `.records.sqlalch`
Rename the database modules to:
* `mysql`
* `pgsql`
* `sqlite3`

**Comment by eddowh** [comment]
Fixed this

**Pull request merged**